// auto-generated by yaml_to_cxx.py from test/upstream/regression/2766.yaml
#include "testlib.h"
void test_upstream_regression_2766() {
    enter_section("test_upstream_regression_2766: Stop people treating ptypes as objects");
    TEST_EQ(maybe_run(R::now()["epoch_time"], *conn), (err("ReqlQueryLogicError", "Cannot call `bracket` on objects of type `PTYPE<TIME>`.")));
    TEST_EQ(maybe_run(R::now().get_field("epoch_time"), *conn), (err("ReqlQueryLogicError", "Cannot call `get_field` on objects of type `PTYPE<TIME>`.")));
    TEST_EQ(maybe_run(R::now().keys(), *conn), (err("ReqlQueryLogicError", "Cannot call `keys` on objects of type `PTYPE<TIME>`.")));
    TEST_EQ(maybe_run(R::now().pluck("epoch_time"), *conn), (err("ReqlQueryLogicError", "Cannot call `pluck` on objects of type `PTYPE<TIME>`.")));
    TEST_EQ(maybe_run(R::now().without("epoch_time"), *conn), (err("ReqlQueryLogicError", "Cannot call `without` on objects of type `PTYPE<TIME>`.")));
    TEST_EQ(maybe_run(R::now().merge(R::object("foo", 4)), *conn), (err("ReqlQueryLogicError", "Cannot call `merge` on objects of type `PTYPE<TIME>`.")));
    TEST_EQ(maybe_run(R::expr(R::object("foo", 4)).merge(R::now()), *conn), (err("ReqlQueryLogicError", "Cannot merge objects of type `PTYPE<TIME>`.")));
    TEST_EQ(maybe_run(R::now().has_fields("epoch_time"), *conn), (err("ReqlQueryLogicError", "Cannot call `has_fields` on objects of type `PTYPE<TIME>`.")));
    TEST_EQ(maybe_run(R::object().has_fields(R::time(2014, 7, 7, "Z")), *conn), (err("ReqlQueryLogicError", "Invalid path argument `1404691200`.")));
    TEST_EQ(maybe_run(R::expr(1).keys(), *conn), (err("ReqlQueryLogicError", "Cannot call `keys` on objects of type `NUMBER`.")));
    exit_section();
}
